source/common/grpc/common.cc

```diff
@@ -138,7 +138,8 @@ Buffer::InstancePtr Common::serializeToGrpcFrame(const Protobuf::Message& messag
   uint8_t* current = reinterpret_cast<uint8_t*>(iovec.mem_);
   *current++ = 0; // flags
   const uint32_t nsize = htonl(size);
-  std::memcpy(current, reinterpret_cast<const void*>(&nsize), sizeof(uint32_t));
+  std::memcpy(current, reinterpret_cast<const void*>(&nsize), // NOLINT(safe-memcpy)
```

Comment:
I think you can instead use
```
SAFE_MEMCPY(reinterpret_cast<uint32_t*>current, nsize);
```
but we'll have to see whether the various sanitizers like that, as `current` is likely not aligned safely for 32-bit operations. I think that's OK though as you are not really asking the compiler to do 32-bit operations. But I wonder whether the compiler may feel justified in transforming
```
uint32_t a, b;
memcpy(&a, &b, sizeof(uint32_t));
```
into
```
uint32_t a, b;
a = b;
```
it's worth a try and also see if any other @envoyproxy/maintainers have enough C++ compiler semantics memorized to opine on whether a compiler should be allowed to do that.

Reply:
Actually https://en.cppreference.com/w/cpp/string/byte/memcpy calls out this case explicitly as a valid use of memcpy. Compilers shouldn't convert to the unsafe 32-bit operation on data they don't know to be aligned.

So my SAFE_MEMCPY suggestion here should be correct.

Reply:
Interesting, so, I will try it now 👍🏻.